pennylane/capture/autograph/transformer.py

```diff
@@ -0,0 +1,227 @@
+
+        # Converted functions are cached as a _PythonFnFactory object.
+        if self._cache.has(fn, TOPLEVEL_OPTIONS):
+            cached_factory = self._cached_factory(fn, TOPLEVEL_OPTIONS)
+        elif self._cache.has(fn, NESTED_OPTIONS):
+            cached_factory = self._cached_factory(fn, NESTED_OPTIONS)
+        else:
+            cached_factory = self._cached_factory(fn, STANDARD_OPTIONS)
+
+        # Convert to a Python function object before returning (e.g. to obtain its source code).
+        new_fn = cached_factory.instantiate(
+            fn.__globals__,
+            fn.__closure__ or (),
+            defaults=fn.__defaults__,
+            kwdefaults=getattr(fn, "__kwdefaults__", None),
+        )
+
+        return new_fn
+
+
+def run_autograph(fn):
+    """Decorator that converts the given function into graph form.
+
+    .. warning::
+
+        Nested functions are only lazily converted by AutoGraph. If the input includes nested
+        functions, these won't be converted until the first time the function is traced.
+    """
+
+    user_context = converter.ProgramContext(TOPLEVEL_OPTIONS)
+
+    new_fn, module, source_map = TRANSFORMER.transform(fn, user_context)
+    new_fn.ag_module = module
+    new_fn.ag_source_map = source_map
+    new_fn.ag_unconverted = fn
+
+    return new_fn
+
+
+def autograph_source(fn):
+    """Utility function to retrieve the source code of a function converted by AutoGraph.
+
+    .. warning::
+
+        Nested functions are only lazily converted by AutoGraph. Make sure that the function has
+        been traced at least once before accessing its transformed source code, for example by
+        specifying the signature of the compiled program or by running it at least once.
+
+    Args:
+        fn (Callable): the original function object that was converted
+
+    Returns:
+        str: the source code of the converted function
+
+    Raises:
+        AutoGraphError: If the given function was not converted by AutoGraph, an error will be
+                        raised.
+
+    **Example**
+
+    .. code-block:: python
+
+        def decide(x):
+            if x < 5:
+                y = 15
+            else:
+                y = 1
+            return y
+
+        @qjit(autograph=True)
+        def func(x: int):
+            y = decide(x)
+            return y ** 2
+
+    >>> print(autograph_source(decide))
+    def decide_1(x):
+        with ag__.FunctionScope('decide', 'fscope', ag__.STD) as fscope:
+            def get_state():
+                return (y,)
+            def set_state(vars_):
+                nonlocal y
+                (y,) = vars_
+            def if_body():
+                nonlocal y
+                y = 15
+            def else_body():
+                nonlocal y
+                y = 1
+            y = ag__.Undefined('y')
+            ag__.if_stmt(x < 5, if_body, else_body, get_state, set_state, ('y',), 1)
+            return y
+    """
```

Comment:
Might make sense to replace this with an example that doesn't use `qjit`.

Reply:
updated to just use `run_autograph` directly